joomla/joomla-cms
# Text separator with menu item params

Pull Request for Issue #16198.

### Summary of Changes
Added setting to choose whether the user wants the "separator" type menu item to be displayed as standard separator (horizontal line) or labelled separator (with title text).

For B/C the titles containing only dashes and spaces will still be displayed as standard separator this can be changed in 4.0 if deemed fit. This note is included in the field description tip.


### Testing Instructions
* Apply patch
* The menu item separators should display exactly the same as before this patch. Check separators under "menu" dropdown.
* Create a new custom menu for backend and add some menu items. Also add separators between items.
* When creating separator menu items, you should be able to enter any title for it and choose whether to use title as label for the separator or not.
* Switch to this custom menu in menu module settings. Your menu should be displayed correctly with appropriate type of separators.

![image](https://user-images.githubusercontent.com/6706189/30511998-83a6be36-9b02-11e7-8aee-9c210f870c18.png)


### Documentation Changes Required
Need to add information about using separator menu items in admin menu manager.

Fixes #16198.
